photon-core/src/main/java/org/photonvision/vision/pipeline/ArucoPipeline.java

```diff
@@ -89,17 +87,18 @@ protected void setPipeParamsImpl() {
     @Override
     protected CVPipelineResult process(Frame frame, ArucoPipelineSettings settings) {
         long sumPipeNanosElapsed = 0L;
-        Mat rawInputMat;
-        rawInputMat = frame.colorImage.getMat();
 
         List<TrackedTarget> targetList;
-        CVPipeResult<List<ArucoDetectionResult>> tagDetectionPipeResult;
 
-        if (rawInputMat.empty()) {
-            return new CVPipelineResult(sumPipeNanosElapsed, 0, List.of(), frame);
+        if (frame.type != FrameThresholdType.GREYSCALE) {
+            // TODO so all cameras should give us ADAPTIVE_THRESH -- how should we handle if not?
```

Comment:
Is this TODO accurate? I'm not positive if it is

Reply:
I copied it over from apriltag, not familiar with how the requested threshold type can fail.

Reply:
yeah, that was a note to myself. Lemme verify

Reply:
It should say GREYSCALE in the TODO. can you change the comment to:

`// We asked for a GREYSCALE frame, but didn't get one -- best we can do is give up`